### ISSUE TITLE
notification plugin installation failed

### ISSUE THREAD

**Issue by olaf7** [scm-manager/scm-manager#1235]
SCM-manager 2.2.0 Debian package on Buster (stable).
Now (inline) migration works for me including service restart. Thanks!
Next was installing plugins. I did this the lazy way: selecting a bunch of plugins to install instead of doing it one by one.

Side note: I did not notice any complaints when migrating although most of my projects depended on the Redmine plugin which was no longer available after migrating. I assume I will need to check all projects for URLs/links/plugins once SCM-manager runs again (now 500 all over the place). 

I get a timeout in the GUI and when continuing a HTTP 500.
Log:

```
2020-07-03 13:33:17.487 [main] [          ] INFO  sonia.scm.plugin.PluginProcessor - collect plugins
2020-07-03 13:33:18.068 [main] [          ] INFO  sonia.scm.plugin.PluginProcessor - install plugin tree:
+- scm-readme-plugin
+- scm-notify-plugin
+- scm-activity-plugin
+- scm-mail-plugin
   +- scm-notify-plugin
   +- scm-review-plugin
+- scm-review-plugin
+- scm-legacy-plugin
+- scm-svn-plugin
+- scm-hg-plugin
+- scm-pushlog-plugin
+- scm-redmine-plugin
+- scm-git-plugin
+- scm-issuetracker-plugin
   +- scm-redmine-plugin

2020-07-03 13:33:19.044 [main] [          ] INFO  sonia.scm.update.repository.XmlRepositoryV1UpdateStep - no v1 repositories database file found
2020-07-03 13:33:19.347 [main] [          ] INFO  sonia.scm.update.UpdateEngine - running update step for type sonia.scm.redmine.config.global.xml and version 2.0.0 (class sonia.scm.redmine.update.RedmineV2GlobalConfigMigrationUpdateStep)
2020-07-03 13:33:19.399 [main] [          ] INFO  sonia.scm.update.UpdateEngine - running update step for type sonia.scm.notify.repository.config.xml and version 2.0.0 (class sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep)
2020-07-03 13:33:19.425 [main] [          ] ERROR sonia.scm.migration.UpdateException - could not execute update for type sonia.scm.notify.repository.config.xml to version 2.0.0 in class sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep
java.lang.NumberFormatException: For input string: ""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:662)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep.buildConfig(NotifyV2RepositoryConfigMigrationUpdateStep.java:89)
	at sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep.lambda$doUpdate$0(NotifyV2RepositoryConfigMigrationUpdateStep.java:73)
	at sonia.scm.update.MapBasedPropertyReaderInstance.call(MapBasedPropertyReaderInstance.java:58)
	at sonia.scm.update.MapBasedPropertyReaderInstance.lambda$forEachEntry$0(MapBasedPropertyReaderInstance.java:44)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet.lambda$entryConsumer$0(Collections.java:1576)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1746)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet$UnmodifiableEntrySetSpliterator.forEachRemaining(Collections.java:1601)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.MapBasedPropertyReaderInstance.forEachEntry(MapBasedPropertyReaderInstance.java:44)
	at sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep.doUpdate(NotifyV2RepositoryConfigMigrationUpdateStep.java:73)
	at sonia.scm.update.UpdateEngine.execute(UpdateEngine.java:87)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.UpdateEngine.update(UpdateEngine.java:77)
	at sonia.scm.lifecycle.BootstrapContextListener.processUpdates(BootstrapContextListener.java:175)
	at sonia.scm.lifecycle.BootstrapContextListener.startEitherMigrationOrApplication(BootstrapContextListener.java:133)
	at sonia.scm.lifecycle.BootstrapContextListener.createMigrationOrNormalModuleProvider(BootstrapContextListener.java:124)
	at sonia.scm.lifecycle.BootstrapContextListener.createStageTwoInjector(BootstrapContextListener.java:117)
	at sonia.scm.lifecycle.BootstrapContextListener.getInjector(BootstrapContextListener.java:93)
	at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:45)
	at sonia.scm.lifecycle.BootstrapContextListener.contextInitialized(BootstrapContextListener.java:75)
	at sonia.scm.lifecycle.BootstrapContextFilter.initializeContext(BootstrapContextFilter.java:74)
	at sonia.scm.lifecycle.BootstrapContextFilter.init(BootstrapContextFilter.java:65)
	at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:134)
	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:749)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:773)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:360)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1445)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1409)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:844)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:275)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.server.Server.start(Server.java:408)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.server.Server.doStart(Server.java:372)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at sonia.scm.server.ScmServer.init(ScmServer.java:130)
	at sonia.scm.server.ScmServer.run(ScmServer.java:91)
	at sonia.scm.server.ScmServerDaemon.main(ScmServerDaemon.java:53)
2020-07-03 13:33:19.425 [main] [          ] ERROR sonia.scm.lifecycle.BootstrapContextListener - failed to create stage two injector
sonia.scm.migration.UpdateException: could not execute update for type sonia.scm.notify.repository.config.xml to version 2.0.0 in class sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep
	at sonia.scm.update.UpdateEngine.execute(UpdateEngine.java:90)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.UpdateEngine.update(UpdateEngine.java:77)
	at sonia.scm.lifecycle.BootstrapContextListener.processUpdates(BootstrapContextListener.java:175)
	at sonia.scm.lifecycle.BootstrapContextListener.startEitherMigrationOrApplication(BootstrapContextListener.java:133)
	at sonia.scm.lifecycle.BootstrapContextListener.createMigrationOrNormalModuleProvider(BootstrapContextListener.java:124)
	at sonia.scm.lifecycle.BootstrapContextListener.createStageTwoInjector(BootstrapContextListener.java:117)
	at sonia.scm.lifecycle.BootstrapContextListener.getInjector(BootstrapContextListener.java:93)
	at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:45)
	at sonia.scm.lifecycle.BootstrapContextListener.contextInitialized(BootstrapContextListener.java:75)
	at sonia.scm.lifecycle.BootstrapContextFilter.initializeContext(BootstrapContextFilter.java:74)
	at sonia.scm.lifecycle.BootstrapContextFilter.init(BootstrapContextFilter.java:65)
	at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:134)
	at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:749)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:734)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:773)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:360)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1445)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1409)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:844)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:275)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.eclipse.jetty.server.Server.start(Server.java:408)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
	at org.eclipse.jetty.server.Server.doStart(Server.java:372)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at sonia.scm.server.ScmServer.init(ScmServer.java:130)
	at sonia.scm.server.ScmServer.run(ScmServer.java:91)
	at sonia.scm.server.ScmServerDaemon.main(ScmServerDaemon.java:53)
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:662)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep.buildConfig(NotifyV2RepositoryConfigMigrationUpdateStep.java:89)
	at sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep.lambda$doUpdate$0(NotifyV2RepositoryConfigMigrationUpdateStep.java:73)
	at sonia.scm.update.MapBasedPropertyReaderInstance.call(MapBasedPropertyReaderInstance.java:58)
	at sonia.scm.update.MapBasedPropertyReaderInstance.lambda$forEachEntry$0(MapBasedPropertyReaderInstance.java:44)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet.lambda$entryConsumer$0(Collections.java:1576)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1746)
	at java.base/java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet$UnmodifiableEntrySetSpliterator.forEachRemaining(Collections.java:1601)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at sonia.scm.update.MapBasedPropertyReaderInstance.forEachEntry(MapBasedPropertyReaderInstance.java:44)
	at sonia.scm.notify.update.NotifyV2RepositoryConfigMigrationUpdateStep.doUpdate(NotifyV2RepositoryConfigMigrationUpdateStep.java:73)
	at sonia.scm.update.UpdateEngine.execute(UpdateEngine.java:87)
	... 45 common frames omitted

```

**Comment by pfeuffer** [comment]
Thanks for the ultra fast feedback :)

Looks like you've stumbled over an issue with the notify update procedure. Somehow we do not check, whether all of the values had been set (in your case the setting for the max diff lines). Can you either

- set all the configuration values for the notify plugin in 1.60 first, or
- do not install the notify plugin in 2.2.0?

We should release a fixed version of the notify plugin, soon.

Regards

René

**Comment by olaf7** [comment]
If I recall correctly:
I had the notification-plugin installed on my old server running 1.60
SCM-home was copied to a new server.
SCM_Manager 2.1.0 was installed, but not running.
I updated to SCM-Manager 2.2.0
Next I inline migrated all projects.
I was able to login to 2.2.0 with old credentials which is ok.
All old projects were listed.
But **NO plugins were installed**. So the plugins were not migrated. Don't know if this is intended behavior.
Next I installed a bunch of plugins. Notify-plugin was/is a dependency and was added to the list by the install process.
I _assumed_ the migration process was already finalized before I installed plugins. I guess I was wrong?

**Comment by pfeuffer** [comment]
Thanks for the clarification. This should be no problem normally. Whenever you first install the plugin in 2.x (no matter if directly after the migration after updating to later versions) it should find its old data and start the upgrade for this subset. The migration process and the update steps of the plugins are not coupled with each other.

**Comment by Pilopa** [comment]
We have resolved a migration issue related to the NumberFormatException you were experiencing in this Pull Request:
https://github.com/scm-manager/scm-notify-plugin/pull/5

**Comment by sdorra** [comment]
The bug should be fixed with version 2.0.1 of the scm-notifiy-plugin.
